oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -320,7 +322,7 @@ private static URL toURL(String url) {
 			@Override
 			public Resource retrieveResource(URL url) throws IOException {
 				HttpHeaders headers = new HttpHeaders();
-				headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON_UTF8));
+				headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, APPLICATION_JWK_SET_JSON));
```

Comment:
Can you also please add a test in `NimbusJwtDecoderTests` that asserts these 2 headers are sent in the request. The test should look similar to `DefaultOAuth2UserServiceTests.loadUserWhenUserInfoSuccessResponseThenAcceptHeaderJson()` making use of `MockWebServer`

Reply:
I have added the requested test.

Reply:
I replaced the test with your proposed technique. But I have doubts if the new one is better. I prefer the old one, because it knows less of the internals of the NimbusJwtDecoder. So the new one is too tightly coupled for my taste. So whenever the implementation changes (like from 5.0 to 5.1) the test has to be rewritten. This is not the case with the old unit test.

But long story short: I can live with both, so it is your call. 🙂